### PR TITLE
apply —lenient to reports emitted while running.

### PR DESCRIPTION
**Pull request by aaroncrespo**
didn't know that some reporters run before all results are gathered. so we apply --lenient at the point of collection per file before `report()` of instead of after when they are aggregated.

### CODE REVIEW

**Comment by aaroncrespo** [comment]
@marcelofabri This is unrelated to Xcode 8.3

**Comment by NachoSoto** [comment]
Is this good to go? :)

File: CHANGELOG.md [hunk 1]
```diff
@@ -112,6 +112,10 @@
 
 ##### Bug Fixes
 
+* Fix `--lenient` reports errors before lint completion.
```
Comment (marcelofabri) [inline CHANGELOG.md]: this is in the wrong section

**Comment by marcelofabri** [comment]
@NachoSoto would you mind testing this for your use case before we merge?

**Comment by jpsim** [comment]
Thanks! Merged to master in bc7fa329.

**Pull request closed without merging**